jpsonic-main/src/main/java/com/tesshu/jpsonic/domain/PlayQueue.java

```diff
@@ -94,24 +95,28 @@ public void setName(String name) {
      * @return The current song in the playlist, or <code>null</code> if no current song exists.
      */
     public MediaFile getCurrentFile() {
-        synchronized (statusLock) {
-            synchronized (sequenceLock) {
-                if (index == -1 || index == 0 && size() == 0) {
-                    setStatus(Status.STOPPED);
-                    return null;
-                } else {
+        readLock(sequenceLock);
+        try {
+            if (index == -1 || index == 0 && size() == 0) {
+                setStatus(Status.STOPPED);
+                return null;
+            } else {
```

Comment:
Is this code necessary or unnecessary?

Reply:
ah,
This is strange code if we want to ensure only the atmic and the visibility, of the "List<MediaFile> files". However, there are cases where the status is changed after checking the existence of the file.

Whatever this behavior is, it may be best not to change it. But, getCurrentFile() requires a writeLock, not a readLock. (While this method is called, inquiries about its status should wait.)

Reply:
Done.